lib/version-handler.js

```diff
@@ -1,6 +1,6 @@
 'use strict';
 
-const fromNpmVersionRegex = /^([@\w-./]+)@[~^]?([\d.]+)/;
+const fromNpmVersionRegex = /^([@\w-./]+)@[~^]?([\d.\-?]+)/;
```

Comment:
@danbev if I understand this correctly, then there are still a few semver cases that it doesn't cover. See http://semver.org/ for the full spec.

Some examples that I tried (taken from the examples on that page) which don't get matched correctly using this regex:

```js
fromNpmVersionRegex.exec('fh-thing@1.0.0-x.7.z.92')
[ 'fh-thing@1.0.0-',
  'fh-thing',
  '1.0.0-',
  index: 0,
  input: 'fh-thing@1.0.0-x.7.z.92' ]

fromNpmVersionRegex.exec('fh-thing@1.0.0-alpha+001')
[ 'fh-thing@1.0.0-',
  'fh-thing',
  '1.0.0-',
  index: 0,
  input: 'fh-thing@1.0.0-alpha+001' ]

fromNpmVersionRegex.exec('fh-thing@1.0.0+20130313144700')
[ 'fh-thing@1.0.0',
  'fh-thing',
  '1.0.0',
  index: 0,
  input: 'fh-thing@1.0.0+20130313144700' ]
```

There might be other examples there too.

Reply:
Thanks Ger, I've taken a look and tried to simplify this.